### PR TITLE
OAK-8962 | Restructuring oak-benchmarks to avoid lucene/solr/elastic …

### PR DESCRIPTION
**Pull request by nit0906**
…conflicts

### CODE REVIEW

File: oak-benchmarks-lucene/src/main/java/org/apache/jackrabbit/oak/benchmark/util/OakLuceneIndexUtils.java [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+            }
+            if (!orderedProps.isEmpty()) {
+                indexDef.setProperty(FulltextIndexConstants.ORDERED_PROP_NAMES,
+                        orderedProps.toArray(new String[orderedProps.size()]),
+                        PropertyType.NAME);
+            }
+        }
+
+        // Set file persistence if specified
+        if (!Strings.isNullOrEmpty(persistencePath)) {
```
Comment (thomasmueller) [inline oak-benchmarks-lucene/src/main/java/org/apache/jackrabbit/oak/benchmark/util/OakLuceneIndexUtils.java:95]: I don't think "persistence" is needed any longer. I would remove all three (PERSISTENCE_PATH, PERSISTENCE_NAME, PERSISTENCE_FILE) and the respective parameter

Comment (nit0906) [reply oak-benchmarks-lucene/src/main/java/org/apache/jackrabbit/oak/benchmark/util/OakLuceneIndexUtils.java:95]: @thomasmueller  - 
In the scalability tests , I see there are specific cases where this is being used - 
```
case LUCENE_FILE:
                persistencePath =
                    "target" + StandardSystemProperty.FILE_SEPARATOR.value() + "lucene" + String
                        .valueOf(System.currentTimeMillis());
                OakLuceneIndexUtils.luceneIndexDefinition(session, "customIndex", ASYNC_INDEX,
                        new String[]{FILTER_PROP, DATE_PROP},
                        new String[]{PropertyType.TYPENAME_STRING, PropertyType.TYPENAME_DATE},
                        null, persistencePath);
                break;
            case LUCENE_FILE_DOC:
                persistencePath =
                    "target" + StandardSystemProperty.FILE_SEPARATOR.value() + "lucene" + String
                        .valueOf(System.currentTimeMillis());
```

I think let's keep it as it is for now ...when I would be setting up all these tests in jenkins, then I can revisit and see if we can safely remove it. Wdyt ?

File: oak-benchmarks-lucene/src/main/java/org/apache/jackrabbit/oak/benchmark/LuceneBenchmarkRunner.java [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+
+import java.util.ArrayList;
+import java.util.Arrays;
+
+public class LuceneBenchmarkRunner extends BenchmarkRunner {
+
+    public static void main(String[] args) throws Exception {
+        statsProvider = options.has(benchmarkOptions.getMetrics()) ? getStatsProvider() : StatisticsProvider.NOOP;
+        initOptionSet(args);
+        BenchmarkRunner.addToBenchMarkList(new ArrayList<>(
```
Comment (fabriziofortino) [inline oak-benchmarks-lucene/src/main/java/org/apache/jackrabbit/oak/benchmark/LuceneBenchmarkRunner.java]: `new ArrayList<>` is not needed

File: oak-benchmarks-lucene/src/main/java/org/apache/jackrabbit/oak/run/BenchMarkLuceneCommand.java [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.jackrabbit.oak.run;
+
+
+import org.apache.jackrabbit.oak.benchmark.LuceneBenchmarkRunner;
+import org.apache.jackrabbit.oak.run.commons.Command;
+
+public class BenchMarkLuceneCommand implements Command {
```
Comment (fabriziofortino) [inline oak-benchmarks-lucene/src/main/java/org/apache/jackrabbit/oak/run/BenchMarkLuceneCommand.java]: for consistency, I would rename this to `BenchmarkLuceneCommand `

File: oak-benchmarks-lucene/src/main/java/org/apache/jackrabbit/oak/run/LuceneMain.java [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+    private static final Modes MODES = new Modes(ImmutableMap.<String, Command>of(
+            "benchmark", new BenchMarkLuceneCommand(),
+            "scalability", new ScalabilityCommand()
+    ));
+
+    private LuceneMain() {
+        // Prevent instantiation.
+    }
+
+    public static void main(String args[]) throws Exception{
```
Comment (fabriziofortino) [inline oak-benchmarks-lucene/src/main/java/org/apache/jackrabbit/oak/run/LuceneMain.java]: nitpick: the syntax `String[] args` is more idiomatic

File: oak-benchmarks-lucene/src/main/java/org/apache/jackrabbit/oak/scalability/LuceneScalabilityRunner.java [hunk 5]
```diff
@@ -0,0 +1,88 @@
...
+import org.apache.jackrabbit.oak.scalability.benchmarks.search.SplitOrderBySearcher;
+import org.apache.jackrabbit.oak.scalability.suites.ScalabilityBlobSearchSuite;
+import org.apache.jackrabbit.oak.scalability.suites.ScalabilityNodeRelationshipSuite;
+import org.apache.jackrabbit.oak.scalability.suites.ScalabilityNodeSuite;
+
+import java.util.Arrays;
+
+public class LuceneScalabilityRunner extends ScalabilityRunner {
+
+    public static void main(String args[]) throws Exception {
```
Comment (fabriziofortino) [inline oak-benchmarks-lucene/src/main/java/org/apache/jackrabbit/oak/scalability/LuceneScalabilityRunner.java]: nitpick: the syntax `String[] args` is more idiomatic

**Review by fabriziofortino — APPROVED** [review]
LGTM, just some minors.

**Pull request closed without merging**